<!-- -*- xml -*- -->

<launch>

<!--node pkg="nmea_navsat_driver" type="nmea_serial_driver" name="nmea_serial_driver" respawn="true">
  <param name="port" type="string" value="/dev/ttyACM0"/>  
  <param name="baud" type="string" value="115200"/> 
  <remap from="/fix" to="/gps/fix" /> 
</node-->

<arg name="map_file" default="/home/vigmesh/maps/polytunnels.yaml"/>

<!-- Run the map server -->
<node name="map_server" pkg="map_server" type="map_server" args="$(arg map_file)" />

<node pkg="amcl" type="amcl" name="amcl" output="screen">

 <remap from="scan" to="scan"/>

    <!-- see param description on http://www.ros.org/wiki/amcl -->     
		<!--overall filter -->
        <param name="min_particles" value="100"/>
        <param name="max_particles" value="5000"/>
        <param name="kld_err" value="0.01"/>
        <param name="kld_z" value="0.99"/>
        <param name="update_min_d" value="0.2"/>
        <param name="update_min_a" value="0.52"/>
        <param name="resample_interval" value="2"/>
        <param name="transform_tolerance" value="0.1"/>
        <param name="recovery_alpha_slow" value="0.0"/>
        <param name="recovery_alpha_fast" value="0.0"/>
        <param name="gui_publish_rate" value="10.0"/>

		<!--initial pose 
        <param name="initial_pose_x" value="12.424"/>
        <param name="initial_pose_y" value="4.182"/>
        <param name="initial_pose_a" value="1.57"/>
        <param name="initial_cov_xx" value="0.0"/>
        <param name="initial_cov_yy" value="0.0"/>
        <param name="initial_cov_aa" value="0.0"/>-->

		<!-- laser model -->
        <param name="laser_max_beams" value="60"/> 
        <param name="laser_z_hit" value="0.95"/>
        <param name="laser_z_short" value="0.1"/>
        <param name="laser_z_max" value="0.05"/>
        <param name="laser_z_rand" value="0.05"/>
        <param name="laser_sigma_hit" value="0.2"/>
        <param name="laser_lambda_short" value="0.1"/>
        <param name="laser_likelihood_max_dist" value="2.0"/>
        <param name="laser_model_type" value="likelihood_field"/>

		<!-- odometry model -->
        <param name="odom_model_type" value="omni-corrected"/>
        <param name="odom_alpha1" value="0.2"/>
        <param name="odom_alpha2" value="0.2"/>
        <param name="odom_alpha3" value="0.2"/>
        <param name="odom_alpha4" value="0.2"/>
        <param name="odom_frame_id" value="odom"/>
        <param name="base_frame_id" value="base_link"/>
        <param name="global_frame_id" value="map" />

		<!-- map -->
        <param name="use_map_topic" value="true"/>

</node>

<node pkg="robot_pose_publisher" type="robot_pose_publisher" name="robot_pose_publisher" output="screen">
 <remap from="robot_pose" to="thorvald_pose"/>
</node>

<!--node pkg="thorvald_2d_nav" type="test_RANSAC" name="test_RANSAC" output="screen"/-->

<!--node pkg="thorvald_2d_nav" type="test_control_modes" name="test_control_modes" output="screen"/-->

<!--node pkg="thorvald_2d_nav" type="fastSLAM" name="fastSLAM" output="screen"/-->

<!--node pkg="thorvald_2d_nav" type="test_row_transition" name="test_row_transition" output="screen"/-->

<!--node pkg="rviz" type="rviz" name="rviz" args="-d $(find thorvald_2d_nav)/rviz/lincoln.rviz"/-->

<!--node pkg="thorvald_2d_nav" type="gps_switch_server" name="gps_switch_server"/-->

</launch>
